### ISSUE TITLE
Add a route to retrieve contacts

### ISSUE THREAD

**Issue by fabrahaingo** [fabrahaingo/find-shelter#7]
This route could be called `/api/contacts/find`

It would by default return all contacts, and a query string could help filter out specific users:
- find only contacts living in `Ile de France` (French region) => `/api/contacts/find?region=ile-de-france`
- find only the user called John Doe => `/api/contacts/find?firstName=john&lastName=doe`
- etc...

The tricky part for this would is security. How to ensuite that only authorized users access this data ?
Isn't it easier to download the `json` file from the db and filter locally on the computer ?

**Comment by petrussola** [comment]
Good idea. But for the address, should the contact form address field be split into multiple inputs i.e. address, ville, region, etc.? Some of these could be drop down, like the region. I suppose we can always filter by the text typed by the user w regex or sth like that, but it may be hit and miss. wdyt?

As for security, I am far from being an expert. Maybe we could build a small admin interface with login for yourself only where these data can be accessed? I think 3rd parties like 0Auth or Firebase make it easy to have a signup/login up and running.

**Comment by fabrahaingo** [comment]
Regarding the address input, @elliotwutingfeng is currently working on this feature in [this issue](https://github.com/fabrahaingo/find-shelter/issues/4) :)
Once done, we'll be able to get a list of suggestions from a user query, exactly like on Algolia Place [try it out here](https://community.algolia.com/places/)

Regarding the admin account, I find it a little overkill if that's only going to be used by me. I can easily access it from my MongoDB Atlas account...
I was more thinking about a solution for the associations I give contacts to. Do I simply send them an Excel sheet containing all contact info ? I don't think it's GDPR-friendly 😞

**Comment by fabrahaingo** [comment]
Maybe sending them a link that would do the following things is okay:
1. give them access to a contact list from a limited time
2. eventually filter the list by certain criteria (like `all contacts living in Ile-de-France` only)
wdyt ?

**Comment by petrussola** [comment]
What about adding a checkbox at the end of the form asking for permission to share contact details with NGO XYZ?
